src/lib.rs

```diff
@@ -363,16 +366,15 @@ impl<const N: usize, T: Gene> System<N, T> {
 
     fn migrate_agents(&mut self) {
         let mut rng = thread_rng();
-        let len = self.islands.len();
-
         let mut push_queue = Vec::new();
 
-        for (i, island) in self.islands.iter_mut().enumerate() {
+        for island in self.islands.iter_mut() {
+            let connected_islands = &self.topology[island._id];
+            if connected_islands.is_empty() {
+                continue;
+            }
```

Comment:
doesn't this mean that if an island is not connected to any other island its agents will be put in the migration queue and never removed, thus becoming effectively erased?